internal/services/search/search_service_resource.go

```diff
@@ -161,46 +216,219 @@ func resourceSearchServiceCreateUpdate(d *pluginsdk.ResourceData, meta interface
 		publicNetworkAccess = services.PublicNetworkAccessDisabled
 	}
 
-	expandedIdentity, err := identity.ExpandSystemAssigned(d.Get("identity").([]interface{}))
+	skuName := services.SkuName(d.Get("sku").(string))
+	ipRulesRaw := d.Get("allowed_ips").(*pluginsdk.Set).List()
+	hostingMode := services.HostingMode(d.Get("hosting_mode").(string))
+	cmkEnforcementEnabled := d.Get("cmk_enforcement_enabled").(bool)
+	apiAccessControl := d.Get("api_access_control").(*pluginsdk.Set).List()
+
+	cmkEnforcement := services.SearchEncryptionWithCmkDisabled
+	if cmkEnforcementEnabled {
+		cmkEnforcement = services.SearchEncryptionWithCmkEnabled
+	}
+
+	// NOTE: hosting mode is only valid if the SKU is 'standard3'
+	if skuName != services.SkuNameStandardThree && hostingMode == services.HostingModeHighDensity {
+		return fmt.Errorf("'hosting_mode' can only be defined if the 'sku' field is set to the 'standard3' SKU, got %q", skuName)
```

Comment:
I was told by another reviewer, in a different PR, that they wanted me to use the hard coded strings instead of the constants since it made reading the error string harder to ascertain during the review. Is there a HashiCorp coding style standard for this? This back and forth depending on the reviewer is a bit confusing as a contributor.